src/prefect/tasks/mysql/mysql.py

```diff
@@ -173,13 +178,27 @@ def run(
                 "The 'fetch' parameter must be one of the following - ('one', 'many', 'all')"
             )
 
+        cursor_types = {
+            "cursor": pymysql.cursors.Cursor,
+            "dictcursor": pymysql.cursors.DictCursor
+        }
+
+        if callable(cursor_type):
+            cursor_class = cursor_type
+        else:
+            cursor_class = cursor_types.get(cursor_type.lower())
+        
+        if cursor_class is None or cursor_class not in cursor_types.values():
+            raise TypeError(f"'cursor_type' should be one of ('cursor', 'dictcursor') or the callable equivalent, got {cursor_type}")
```

Comment:
```suggestion
        if not cursor_class:
            raise TypeError(f"'cursor_type' should be one of ('cursor', 'dictcursor') or a Cursor class type, got {cursor_type}")
```

I don't think Jim meant to limit the cursors to just these two. The callable should allow custom cursors to be passed in.

Reply:
Also having this here with the `.get` you wrote is a great way to check for missing types. Nice :)

Reply:
Also note the TypeError can use your dict.keys() to display the possible string cursor values

Reply:
> I don't think Jim meant to limit the cursors to just these two. The callable should allow custom cursors to be passed in.

I'll remove this check. I initially included to avoid potentially confusing or unexpected behavior resulting from the `SS*` cursor classes, as I described here: https://github.com/PrefectHQ/prefect/pull/3574#discussion_r512978435. If a user sets `cursor_type=pymysql.cursors.SSCursor`.

However, I tested against an actual db and it seems to work just fine, even though the current `MySQLFetch` implementation doesn't use it effectively (ideally it would iterate over the cursor and yield rows).

With that said, I think it makes sense to support all of the cursors.